core-services/chat/src/main/java/org/silverpeas/core/chat/servers/EJabberdServer.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2000 - 2016 Silverpeas
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * As a special exception to the terms and conditions of version 3.0 of
+ * the GPL, you may redistribute this Program in connection with Free/Libre
+ * Open Source Software ("FLOSS") applications as described in Silverpeas's
+ * FLOSS exception.  You should have received a copy of the text describing
+ * the FLOSS exception, and it is also available here:
+ * "http://www.silverpeas.org/docs/core/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.chat.servers;
+
+import org.silverpeas.core.admin.user.model.User;
+import org.silverpeas.core.chat.ChatServerException;
+import org.silverpeas.core.chat.ChatUser;
+import org.silverpeas.core.util.JSONCodec;
+import org.silverpeas.core.util.SettingBundle;
+import org.silverpeas.core.util.logging.SilverLogger;
+
+import javax.ws.rs.core.Response;
+import java.util.function.Function;
+
+/**
+ * Adapter to a remote ejabberd service. It implements the mechanisms to access an ejabberd service
+ * in order to register users and to create networking relationships.
+ * @author mmoquillon
+ */
+@DefaultChatServer
+public class EJabberdServer implements ChatServer {
+
+  private SilverLogger logger = SilverLogger.getLogger(this);
+
+  private final String url;
+
+  private final HttpRequester requester;
+
+  /**
+   * Constructs a new instance.
+   */
+  public EJabberdServer() {
+    SettingBundle settings = ChatServer.getChatSettings();
+    this.url = settings.getString("chat.xmpp.restUrl");
+    String key = settings.getString("chat.xmpp.restKey");
+    this.requester = new HttpRequester("Bearer " + key);
+  }
+
+  @Override
+  public void createUser(final User user) throws ChatServerException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.silverpeas.core.chat.ChatServerException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)